src/api/bundler-namespace.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  UserOperation,
+  UserOperationEstimateGasResponse,
+  UserOperationReceipt,
+  UserOperationRequest
+} from '../types/types';
+import { deepHexlify } from '../util/util';
+import { AlchemyConfig } from './alchemy-config';
+
+/**
+ * The Bundler namespace contains methods used for interacting performing user operations and
+ * checking on the state of submitted operations.
+ *
+ * Do not call this constructor directly. Instead, instantiate an Alchemy object
+ * with `const alchemy = new Alchemy(config)` and then access the transact
+ * namespace via `alchemy.bundler`.
+ */
+export class BundlerNamespace {
+  /** @internal */
+  constructor(private readonly config: AlchemyConfig) {}
+
+  /**
+   * Sends a User Operation
+   *
+   * Sends a user operation to the provided bundler client evm network, and
+   * returns a user operation hash on successful submission. The gas fields, if left empty,
+   * will be estimated prior to submitting the UserOperation
+   *
+   * @param userOperation - The user operation.
+   * @beta
+   */
+  async sendUserOperation(
+    userOperation: UserOperationRequest
+  ): Promise<string> {
+    const provider = await this.config.getProvider();
+    if (
+      !userOperation.callGasLimit ||
+      !userOperation.preVerificationGas ||
+      !userOperation.verificationGasLimit
+    ) {
+      const gasEstimates = await this.estimateUserOperationGas(userOperation);
+      userOperation.callGasLimit = gasEstimates.callGasLimit;
+      userOperation.preVerificationGas = gasEstimates.preVerificationGas;
+      userOperation.verificationGasLimit = gasEstimates.verificationGasLimit;
+    }
+
+    if (!userOperation.maxFeePerGas || !userOperation.maxPriorityFeePerGas) {
+      const feeData = await provider.getFeeData();
+      userOperation.maxFeePerGas = feeData.maxFeePerGas;
+      userOperation.maxPriorityFeePerGas = feeData.maxPriorityFeePerGas;
+    }
```

Comment:
ah h/u this is my bad, we can't do this because it changes the hash which will impact the signature